lightningd/chaintopology.c

```diff
@@ -364,6 +365,21 @@ static void topo_update_spends(struct chain_topology *topo, struct block *b)
 	}
 }
 
+static void topo_add_utxos(struct chain_topology *topo, struct block *b)
+{
+	for (size_t i = 0; i < tal_count(b->full_txs); i++) {
+		const struct bitcoin_tx *tx = b->full_txs[i];
+		for (size_t j = 0; j < tal_count(tx->output); j++) {
+			const struct bitcoin_tx_output *output = &tx->output[j];
+			if (is_p2wsh(output->script, NULL)) {
+				wallet_utxoset_add(topo->wallet, tx, j,
+						   b->height, i, output->script,
+						   output->amount);
+			}
+		}
+	}
+}
+
```

Comment:
Is this a wise choice?  That may be putting a lot of data into the db...

Reply:
It only stores P2WSH outputs, which may actually be funding txs. The cost is about 200 bytes / output so without pruning spent ones all P2WSH outputs since segwit was activated take about 30MB in size, much less with pruning. It's the usual space-vs-speed tradeoff and this way we can run on pruned nodes and not parse blocks over and over again.